lib/nodetypes/distributor.rb

```diff
@@ -47,13 +58,17 @@ def is_synced?
   # Should only be called on a distributor that has finished its regular ideal state
   # operations.
   def count_ideal_replicas_not_ready
-    non_ready = 0
+    get_ideal_replicas_not_ready().size/2
+  end
+
+  def get_ideal_replicas_not_ready
+    non_ready = []
     each_database_bucket do |space, bucket_id, state, raw_state|
       # Output is in ideal state order, so expect the 1st one to always be ready.
       # This should generalize to always be true for indexed, store-only, flat and grouped setups.
       # We'd ideally check the active-flag as well, but clusters without indexed docs will
       # not activate any replicas.
-      non_ready += 1 if not state[0].ready
+      non_ready.push(bucket_id, state) if not state[0].ready
```

Comment:
Consider pushing as an array "tuple" `[bucket_id, state]` instead to have each entry take up one array slot. Avoids having to always remember to divide by the expected number of entries per non-ready bucket, and makes it easier to associate each bucket with its state.

Reply:
Fixed